test/dag_benchmarks.cc

```diff
@@ -74,9 +75,9 @@ TEST_CASE("benchmark task handling", "[benchmark][group:task-graph]") {
 			// create simplest possible host task
 			const auto highest_tid = tm.submit_command_group([](handler& cgh) { cgh.host_task(on_master_node, [] {}); });
 			// start notifying once we've built some tasks
-			if(i % report_interval == 0 && i / report_interval > 1) {
+			if(i % report_interval == 0 && i / report_interval > 2) {
 				// every other generated task is always a horizon (step size 0)
-				tm.notify_horizon_reached(highest_tid + 1);
+				tdag.erase_before_epoch(highest_tid - 1);
```

Comment:
I don't quite follow this change

Reply:
Well, this test has a horizon step size of 1, so every other task is a horizon. Reaching horizon $t_{n+1}$ means epoch $t_{n-1}$ has been applied. Previously `task_manager` did duplicate the "the last horizon becomes an epoch" logic itself, but this function is now gone and we need to erase from the TDAG manually.

There is more than one test that expects behavior like this, and it's probably right to ask if they're still testing anything meaningful that I'm not just writing out manually in the tests after the refactoring.